app/src/app/components/specmath-modal/modal.component.ts

```diff
@@ -15,50 +15,58 @@
 import { Component, OnInit } from '@angular/core';
 import { MatDialogRef } from '@angular/material/dialog';
 import { MatStepper } from '@angular/material/stepper';
-import { SpecNameInputOptions, DefaultsFileUploadOptions } from 'src/shared/interfaces';
+import { SpecNameInputOptions, DefaultsFileUploadOptions, SpecFilesUploadOptions } from 'src/shared/interfaces';
 
-type FileUpload = 'default' | 'spec';
+const toolTipText = {
+  1: 'You must name your new spec',
+  3: 'You must upload a set of spec files'
+};
 
 @Component({
   selector: 'app-modal',
   templateUrl: './modal.component.html',
   styleUrls: ['./modal.component.scss']
 })
 export class ModalComponent implements OnInit {
-  minSteps: number;
+  MAX_STEPS = 4;
+  MIN_STEPS = 1;
   currentStep: number;
-  maxSteps: number;
-  specNameInputOptions?: SpecNameInputOptions;
-  defaultsFileUploadOptions?: DefaultsFileUploadOptions;
+  specNameInputOptions: SpecNameInputOptions;
+  defaultsFileUploadOptions: DefaultsFileUploadOptions;
+  specFilesUploadOptions: SpecFilesUploadOptions;
 
   constructor(readonly dialogRef: MatDialogRef<ModalComponent>) {
     dialogRef.disableClose = true;
   }
 
-  get newFileName(): string {
-    if (!this.specNameInputOptions) {
-      return '';
-    }
-
-    return this.specNameInputOptions.newFileName;
-  }
-
   nextStep(stepper: MatStepper): void {
-    if (this.currentStep < this.maxSteps) {
+    if (this.currentStep < this.MAX_STEPS) {
       this.currentStep++;
       stepper.next();
     }
   }
 
   previousStep(stepper: MatStepper): void {
-    if (this.currentStep > this.minSteps) {
+    if (this.currentStep > this.MIN_STEPS) {
       this.currentStep--;
       stepper.previous();
     }
   }
 
-  get specNameValid(): boolean {
-    return !!this.specNameInputOptions?.valid;
+  get newFileName(): string {
+    return (this.specNameInputOptions ? this.specNameInputOptions.newFileName : '');
```

Comment:
One liner is much better. Parentheses should not be needed here.
Also consider `return this.specNameInputOptions?.newFileName || '';`

Reply:
Updated